### ISSUE TITLE
[Feature Request] - Option to create top level categories or projects

### ISSUE THREAD

**Issue by a575606** [nextcloud/tasks#287]
I'm very happy having moved all my contacts, calendars, etc. to Nextcloud. I really like how Tasks links with the calendar reminders and would love to start using the app full time. But there's one feature missing for me which is making it hard to give up Todoist, and that's Projects or Categories. 

Is there any chance of adding some sort of grouping to task lists similar todoists implementation? I can create multiple task lists and create some groupings, but there doesn't seem an easy way to move tasks between lists, so it's not ideal for my usage. 

Other than that, really like the work you've done.

**Comment by raimund-schluessler** [comment]
> Is there any chance of adding some sort of grouping to task lists similar todoists implementation?

I don't know how todoist does that. Could you give a bit more details, an example or a screenshot?

> but there doesn't seem an easy way to move tasks between lists,

You can use drag and drop to move tasks between lists. The problem is, that dnd is apparently broken on Chrome at the moment. See https://github.com/nextcloud/tasks/issues/272#issuecomment-463318338 for more details.

**Comment by a575606** [comment]
Sure, here's a shot. Todoist uses Projects, which allows tasks to be broken up into groups. They can also have sub projects, so it allows task lists to be very neatly organized. That makes it easy to have day to day task, shopping lists and long term projects together without becoming a mess. 
![todoist](https://user-images.githubusercontent.com/11908864/53017187-fb1e4000-3481-11e9-8551-0c890706976f.png)

**Comment by raimund-schluessler** [comment]
Ok, so it is like having nested lists or folders with multiple other lists inside. I think this is a duplicate of https://github.com/nextcloud/tasks/issues/23 then ;)

Speak up if I got it wrong.

**Comment by a575606** [comment]
I think in essence it's the same problem described in  #220. The need for a way to better organize tasks into categories.

**Comment by a575606** [comment]
I suppose it's a similar concept to #23, yes and I think a few others, all different solutions to the same problem. However, the fact that was posted in 2016 isn't encouraging! Is there any chance this could become a priority any time soon? It's important enough to me I was even thinking of dusting off my coding skills to see if I could contribute in any way.

**Comment by raimund-schluessler** [comment]
Well, the top priority was/is to get rid of AngularJS and release the Vue.js rewrite of the app (see https://github.com/nextcloud/tasks/milestone/1 for what is left to do for that).

Once this is done, I am open to implementing new features. I will most likely select them by how much they are demanded, which means repeating tasks and sharing lists will probably come first (it also depends a bit how easy it is to implement it ;).

As for folders, the problem will be, that afaik this is not a CalDAV feature. So no other client will be able to see this structure and we would need to introduce a way of storing it to the server.

**Comment by a575606** [comment]
I haven't read the caldav docs but I was beginning to suspect as much. In my mission to become fully self hosted, task management has been the biggest challenge surprisingly. I suppose it makes sense if the features I need aren't part of caldav. I'll try and make do with multiple lists for now, but keeping fingers crossed for the future. 

Just a thought, but if tags are supported, one easy way around implementing this would be using tags to handle groups in the interface?

**Comment by raimund-schluessler** [comment]
> Just a thought, but if tags are supported, one easy way around implementing this would be using tags as a way to maintain folder structure? Just show tags as folders in the interface?

But tags are for tasks, not for lists. This will give another option to sort/organize tasks, but won't quite be what you requested, I think.

**Comment by a575606** [comment]
True and being able to sort and group and organize tasks in the ui based on tags seems a quick fix that would make a huge difference in usability if it was done right

**Comment by a575606** [comment]
Hi. I noticed it's possible to add children to tasks in the interface. However, they don't seem to remain linked when synced to other caldav applications like Thunderbird or Tasks on android. Is this also a limitation of CalDav?

**Comment by raimund-schluessler** [comment]
> Is this also a limitation of CalDav?

This time it is not the fault of CalDAV. We use the `RELATED-TO` property specified here: https://tools.ietf.org/html/rfc5545#section-3.8.4.5
Thunderbird and Tasks on Android just don't support showing parent-child relations and will show simply all tasks in a list ignoring `RELATED-TO`.
